### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by arianna-h-h**

### CODE REVIEW

File: gistUpload.test.js [hunk 1]
```diff
@@ -12,10 +12,38 @@ describe('When given a string contents and a string gist', () => {
   });
 });
 
-describe('When given no arguments', () => {
-  it('Returns a 422 error code', () => {
+describe('When given no arguments or incorrect number of arguments', () => {
+  it('Returns a new error message: you must provide a file and name.', () => {
```
Comment (snmgian) [inline gistUpload.test.js]: You don't need to mock `axios.post` for this test

File: gistUpload.test.js [hunk 2]
```diff
@@ -12,10 +12,38 @@ describe('When given a string contents and a string gist', () => {
...
-    gistUpload.uploadFile('hello', 'helloName')
-      .then((result) => { expect(result).toBe('422'); });
+    gistUpload.uploadFile()
+      .catch((error) => {
+        expect(error.message).toEqual('You must provide a file and a name.');
+      });
+  });
+
+  it('Returns a new error message: you must provide a file and name.', () => {
+    axios.post = jest.fn(() => Promise.reject(mockBadResponse));
```
Comment (snmgian) [inline gistUpload.test.js]: idem

File: gistUpload.test.js [hunk 3]
```diff
@@ -12,10 +12,38 @@ describe('When given a string contents and a string gist', () => {
...
+    gistUpload.uploadFile('hello')
+      .catch((error) => {
+        expect(error.message).toEqual('You must provide a file and a name.');
+      });
+  });
+});
+
+describe('When given non-string arguments', () => {
+  it('Returns the error: Your file must contain a string.', () => {
+    axios.post = jest.fn(() => Promise.reject(mockBadResponse));
```
Comment (snmgian) [inline gistUpload.test.js]: idem

File: gistUpload.test.js [hunk 4]
```diff
@@ -12,10 +12,38 @@ describe('When given a string contents and a string gist', () => {
...
+    gistUpload.uploadFile(1)
+      .catch((error) => {
+        expect(error.message).toEqual('Your file must contain a string.');
+      });
+  });
+
+  it('Returns the error: Your gist name must be a string.', () => {
+    axios.post = jest.fn(() => Promise.reject(mockBadResponse));
```
Comment (snmgian) [inline gistUpload.test.js]: idem

**Review by snmgian — CHANGES_REQUESTED** [review]

File: gistUpload.js [hunk 5]
```diff
@@ -13,23 +13,30 @@ const HANG_PERCENT = 0.25;
...
-    .catch(error => (console.log(error.response.data)));
+function uploadFile(contents, gist) {
+  try {
+    if (!gist || !contents) {
+      throw new Error('You must provide a file and a name.');
+    } else if (typeof contents !== 'string') {
+      throw new Error('Your file must contain a string.');
+    } else if (typeof gist !== 'string') {
+      throw new Error('Your gist name must be a string.');
+    } else {
```
Comment (snmgian) [inline gistUpload.js]: Don't have an else here. Just close the if/else blocks. Smth like:

```javascript
if (...) {
  ...
} else if ( ... ) {
  ...
}

const post = { ... }
```

So, you can save an indentation level, which is better for readability

File: gistUpload.js [hunk 6]
```diff
@@ -13,23 +13,30 @@ const HANG_PERCENT = 0.25;
...
+      const post = { [gist]: { content: contents } };
+      return axios.post(
+        API_URL,
+        { files: post },
+      )
+        .then((response) => {
+          console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
+          return (response.data.html_url);
+        })
+        .catch(error => error);
```
Comment (snmgian) [inline gistUpload.js]: Don't discard errors

File: gistUpload.js [hunk 7]
```diff
@@ -13,23 +13,30 @@ const HANG_PERCENT = 0.25;
...
-      password: process.env.SECRET_TOKEN,
-    },
-    data: { files: { [gist]: { content } } },
-  })
-    .then((response) => {
-      console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
-      return (response.data.html_url);
-    })
-    .catch(error => (console.log(error.response.data)));
+function uploadFile(contents, gist) {
```
Comment (snmgian) [inline gistUpload.js]: This should be an async function, and validations should be outside of the `try / catch`. smth like:

```javascript

async function ... {
  if (!gist || ...) {
    throw new Error(...);
  } else if ( ... ) {
    ...
  }

  try {
    const response = await axios.post( ... );

    console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
    return (response.data.html_url);

  } catch (error) {
    // if you want to catch the error here, you'll need the try/catch block ...
  }
}

**Review by snmgian — CHANGES_REQUESTED** [review]

File: gistUpload.js (hunk 5 above)
Comment (arianna-h-h) [reply gistUpload.js]: done

File: gistUpload.js (hunk 7 above)
Comment (arianna-h-h) [reply gistUpload.js]: done

File: gistUpload.js (hunk 6 above)
Comment (arianna-h-h) [reply gistUpload.js]: done

File: uploadFile.js [hunk 8]
```diff
@@ -10,17 +10,25 @@ const API_URL = 'https://api.github.com/gists';
...
-    .catch(error => (error.response.status));
+  try {
+    const response = await axios.post(
+      API_URL,
+      { files: post },
+    );
+    console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
+    return (response.data.html_url);
+  } catch (error) {
+    return Promise.reject(error);
```
Comment (snmgian) [inline uploadFile.js]: If you will just reject with the error then you don't need the try/catch block

**Review by snmgian — CHANGES_REQUESTED** [review]

File: uploadFile.js (hunk 8 above)
Comment (arianna-h-h) [reply uploadFile.js]: done

**Review by snmgian — APPROVED** [review]

**Pull request merged**